src/box/memtx_tx.h

```diff
@@ -539,6 +539,63 @@ memtx_tx_snapshot_clarify(struct memtx_tx_snapshot_cleaner *cleaner,
 	return memtx_tx_snapshot_clarify_slow(cleaner, tuple);
 }
 
+#if defined(ENABLE_READ_VIEW)
+
+/** Helper of memtx_tx_snapshot_invisible_count_matching. */
+size_t
+memtx_tx_snapshot_invisible_count_matching_until_slow(
+	struct memtx_tx_snapshot_cleaner *cleaner, struct index_def *index_def,
+	enum iterator_type type, const char *key, uint32_t part_count,
+	struct tuple *until, hint_t until_hint);
+
+/**
+ * Count the amount of invisible tuples matching to the given @a key and
+ * iterator @a type. The tuples are matched using the key_def.
+ */
+static inline size_t
+memtx_tx_snapshot_invisible_count_matching(
+	struct memtx_tx_snapshot_cleaner *cleaner, struct index_def *index_def,
+	enum iterator_type type, const char *key, uint32_t part_count)
+{
+	if (cleaner->ht == NULL)
+		return 0;
+	return memtx_tx_snapshot_invisible_count_matching_until_slow(
+		cleaner, index_def, type, key, part_count, NULL, HINT_NONE);
+}
+
+/**
+ * Count the amount of invisible tuples matching to the given @a key and
+ * iterator @a type up to the given tuple (exclusively) according to the
+ * index order in the snapshot. The tuples are matched using the key_def,
+ * the @a until tuple is compared using the cmp_def.
+ */
+static inline size_t
+memtx_tx_snapshot_invisible_count_matching_until(
+	struct memtx_tx_snapshot_cleaner *cleaner, struct index_def *index_def,
+	enum iterator_type type, const char *key, uint32_t part_count,
+	struct tuple *until, hint_t until_hint)
+{
+	if (cleaner->ht == NULL)
+		return 0;
+	return memtx_tx_snapshot_invisible_count_matching_until_slow(
+		cleaner, index_def, type, key, part_count, until, until_hint);
+}
+
+/**
+ * Detect whether the key of @a tuple from an index with the given @a key_def
+ * is visible in the snapshot with the given @a cleaner.
+ */
+static inline bool
+memtx_tx_snapshot_tuple_key_is_visible(
+	struct memtx_tx_snapshot_cleaner *cleaner,
+	struct key_def *key_def, struct tuple *tuple)
+{
+	return memtx_tx_snapshot_clarify(cleaner, tuple) != NULL &&
+	       !tuple_key_is_excluded(tuple, key_def, MULTIKEY_NONE);
+}
+
+#endif
```

Comment:
Fixed the comment and Included the `memtx_tx_read_view.c` into `memtx_tx.c` under `#ifdef` and created the file in the EE. This way it can use the `memtx_tx_tuple_matches[_uintil]` utility functions from the `memtx_tx.c` and be only defined in EE.

I'd also include a new `memtx_tx_read_view.h` into `memtx_tx.h` instead of having the declaration and implementation scattered over different repositories. Let me know what you think about it, I can't see anything similar in CE now.